src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,107 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        this.elements = new Object[DEFAULT_CAPACITY];
+        this.size = 0;
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("index <" + index
+                    + "> out of ArrayList current size bounds <" + size + ">");
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        resizeIdNeeded();
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("index <" + index
+                    + "> out of ArrayList current size bounds <" + size + ">");
+        }
+
+        if (elements.length == size) {
+            resizeIdNeeded();
+        }
 
+        for (int i = size - 1; i >= index; i--) {
+            elements[i + 1] = elements[i];
+        }
+        elements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return (T) elements[index];
     }
 
     @Override
     public void set(T value, int index) {
+        checkIndex(index);
+        elements[index] = value;
 
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        T removedElement = (T) elements[index];
+        System.arraycopy(elements, index + 1, elements, index, size - 1 - index);
+        size--;
+        return removedElement;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < this.size; i++) {
+            T currentElement = (T) this.elements[i];
+            if (currentElement == element || currentElement != null
+                    && currentElement.equals(element)) {
+                remove(i);
+                return currentElement;
+            }
+        }
+        throw new NoSuchElementException("Cannot find element <"
+                + (element == null ? "null" : element.toString()) + ">");
```

Comment:
you don't need that ternary check, just leave `element` as is, `toString()` will be called by default, null will be printed as "null" automatically